Common/AR/source/ArWrapper.cpp

```diff
@@ -0,0 +1,109 @@
+/*******************************************************************************
+ *  Copyright 2012-2019 Esri
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ ******************************************************************************/
+
+#include "ArWrapper.h"
+
+using namespace Esri::ArcGISRuntime::Toolkit::Internal;
+
+#ifdef Q_OS_IOS
+
+// implemented in ArKitWrapper
+
+#elif defined Q_OS_ANDROID
+
+// implemented in ArCoreWrapper
+
+#else
+
+// default implementation for desktop platforms.
+
+ArWrapper::ArWrapper(void*)
+{
+}
```

Comment:
For an AR optional app (see my comment about `optional` vs `required` in `AndroidManifest.xml`), I guess it can be interesting to build and run on Desktop. And this can be helpful to debug the scene.

For the implementation of `ArcGISArView`, it's easier to have an `ArWrapper` with the same API than `ArCoreWrapper` and `ArKitWrapper`.

Reply:
That makes sense to be able to prototype and run on Desktop. Let's leave this as-is.